File: kt/main/kotlin/moduleconfig/pushtovar/fs/cfgadapt/FooSflCfgAdapter.kt
package tryout.moduleconfig.pushtovar.fs.cfgadapt

import tryout.moduleconfig.pushtovar.config.AppCfgInfo
import tryout.moduleconfig.pushtovar.fs.FooSflCfgInfo
import tryout.moduleconfig.pushtovar.fs.fooSflCfgSrc
import tryout.moduleconfig.pushtovar.fwk.CfgSrcAdaptation

fun fooSflCfgAdapter(appCfg: AppCfgInfo): FooSflCfgInfo {
	return FooSflCfgInfo(
        x = appCfg.x,
    )
}

val fooSflCfgAdaptation = CfgSrcAdaptation(
	fooSflCfgSrc,
	::fooSflCfgAdapter,
)

File: rs/pullcfgwithoverride/src/fs/foo_sfl_cfg_adapter.rs
use super::FooSflCfgInfo;
use common::config::AppCfgInfo;

pub fn foo_sfl_cfg_adapter(app_cfg: &AppCfgInfo) -> FooSflCfgInfo {
    FooSflCfgInfo {
        x: app_cfg.x.clone(),
    }
}
